dashboard/test/ui/features/acquisition_products/curriculum_catalog.feature

```diff
@@ -160,7 +160,7 @@ Feature: Curriculum Catalog Page
 
     And I click selector "[aria-label='View details about AI for Oceans']"
 
-    And I click selector "a:contains(See curriculum details)"
+    And I click selector "a:contains(See curriculum details)" to load a new page
     And I wait until element "h1:contains(AI for Oceans)" is visible
```

Comment:
Isn't line 164 already doing the waiting for an element? Or does "to load a new page" give us more waiting time?

Reply:
Good question - There is more detailed information on this patter in this PR https://github.com/code-dot-org/code-dot-org/pull/56711

The error `ReferenceError: $ is not defined.` typically means JQuery wasn't loaded in the page when a query was executed. 

Line 164 executes in two steps https://github.com/code-dot-org/code-dot-org/blob/839e16166f6d4538346fcf09368b5a1ce1764a2d/dashboard/test/ui/features/step_definitions/steps.rb#L251C1-L254C4
1. Wait for JQuery to load
2. Wait for the expression, i.e., element visibility check to become true

without `to load a new page` on line 163, the JQuery check can be intermittently executed on the old page and subsequently, the JQuery expression to check element visibility gets executed on the new page before JQuery is loaded resulting in the error